src/utiltime.cpp

```diff
@@ -3,26 +3,85 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#define __STDC_WANT_LIB_EXT1__ 1 // for gmtime_s
+
 #if defined(HAVE_CONFIG_H)
 #include "config/bitcoin-config.h"
 #endif
 
 #include "utiltime.h"
 
+#include <chrono>
+#include <thread>
+#include <locale>
+#include <sstream>
+#include <string.h>
+#include <time.h>
 #include <boost/date_time/posix_time/posix_time.hpp>
 #include <boost/thread.hpp>
 
 using namespace std;
 
 static int64_t nMockTime = 0; //!< For unit testing
 
+static inline const tm gmtime_int(time_t time)
+{
+    tm out = {};
+#if defined(HAVE_W32_GMTIME_S)
+    gmtime_s(&out, &time);
+#elif defined(HAVE_C11_GMTIME_S)
+    gmtime_s(&time, &out);
+#elif defined(HAVE_GMTIME_R)
+    gmtime_r(&time, &out);
+#else
+    // Not thread-safe!
+    out = *gmtime(&time);
+#endif
+    return out;
+}
+
+bool ChronoSanityCheck()
```

Comment:
> Best I can tell from google and reading the c spec, it's not guaranteed to be based on the UNIX epoch either :(

Which ones? The man page of "gmtime" tells me the following:
```
       The  ctime(),  gmtime()  and localtime() functions all take an argument of data type time_t, which represents calendar time.  When interpreted as an absolute time value, it represents the
       number of seconds elapsed since the Epoch, 1970-01-01 00:00:00 +0000 (UTC).
```

If using std::chrono makes anything less hassle I'm fine with using it. It just seems like a lot of code necessary to work around issues (and that's just to get started using it!), but you're right that gettimeofday and strptime have their own issues.